python/cudf/cudf/core/_internals/timezones.py

```diff
@@ -0,0 +1,71 @@
+# Copyright (c) 2023, NVIDIA CORPORATION.
+
+import os
+import zoneinfo
+from functools import lru_cache
+
+from cudf._lib.timezone import make_timezone_transition_table
+from cudf.core.dataframe import DataFrame
+
+
+@lru_cache(maxsize=1024)
+def get_tz_data(zone_name):
+    """
+    Return timezone data (transition times and UTC offsets) for the
+    given IANA time zone.
+
+    Parameters
+    ----------
+    zone_name: str
+        IANA time zone name
+
+    Returns
+    -------
+    DataFrame with two columns containing the transition times ("dt")
+    and corresponding UTC offsets ("offset").
+    """
+    try:
+        # like zoneinfo, we first look in TZPATH
+        return _find_and_read_tzfile_tzpath(zone_name)
+    except zoneinfo.ZoneInfoNotFoundError:
+        # if that fails, we fall back to using `tzdata`
+        return _find_and_read_tzfile_tzdata(zone_name)
+
+
+def _find_and_read_tzfile_tzpath(zone_name):
+    for search_path in zoneinfo.TZPATH:
+        if os.path.isfile(os.path.join(search_path, zone_name)):
+            return _read_tzfile_as_frame(search_path, zone_name)
+    raise zoneinfo.ZoneInfoNotFoundError(zone_name)
+
+
+def _find_and_read_tzfile_tzdata(zone_name):
+    import importlib.resources
+
+    package_base = "tzdata.zoneinfo"
+    try:
+        return _read_tzfile_as_frame(
+            str(importlib.resources.files(package_base)), zone_name
+        )
+    # TODO: make it so that the call to libcudf raises a
+    # FileNotFoundError instead of a RuntimeError
```

Comment:
Is this in scope for this PR? Or a future TODO?

Reply:
I think this is more generally part of https://github.com/rapidsai/cudf/issues/12885 and I'd prefer not to do it in this PR.